stdlib/public/Concurrency/AsyncStreamBackpressure.swift

```diff
@@ -0,0 +1,1961 @@
+        /// first element of the provided sequence. If the asynchronous stream already terminated then this method will throw an error
+        /// indicating the failure.
+        ///
+        /// This method returns once more elements should be produced.
+        ///
+        /// - Parameters:
+        ///   - sequence: The elements to write to the asynchronous stream.
+        @available(SwiftStdlib 9999, *)
+        public func write<S>(contentsOf sequence: S) async throws where Element == S.Element, S: Sequence {
+            let writeResult = try { try self.write(contentsOf: sequence) }()
+
+            switch writeResult {
+            case .produceMore:
+                return
+
+            case .enqueueCallback(let callbackToken):
+                try await withTaskCancellationHandler {
+                    try await withCheckedThrowingContinuation { continuation in
+                        self.enqueueCallback(callbackToken: callbackToken, onProduceMore: { result in
+                            switch result {
+                            case .success():
+                                continuation.resume(returning: ())
+                            case .failure(let error):
+                                continuation.resume(throwing: error)
+                            }
+                        })
+                    }
+                } onCancel: {
+                    self.cancelCallback(callbackToken: callbackToken)
+                }
+            }
+        }
+
+        /// Write new element to the asynchronous stream.
+        ///
+        /// If there is a task consuming the stream and awaiting the next element then the task will get resumed with the
+        /// provided element. If the asynchronous stream already terminated then this method will throw an error
+        /// indicating the failure.
+        ///
+        /// This method returns once more elements should be produced.
+        ///
+        /// - Parameters:
+        ///   - sequence: The element to write to the asynchronous stream.
+        @available(SwiftStdlib 9999, *)
+        public func write(_ element: Element) async throws {
+            try await self.write(contentsOf: CollectionOfOne(element))
+        }
+
+        /// Write the elements of the asynchronous sequence to the asynchronous stream.
+        ///
+        /// This method returns once the provided asynchronous sequence or the  the asynchronous stream finished.
+        ///
+        /// - Important: This method does not finish the source if consuming the upstream sequence terminated.
+        ///
+        /// - Parameters:
+        ///   - sequence: The elements to write to the asynchronous stream.
+        @available(SwiftStdlib 9999, *)
+        public func write<S>(contentsOf sequence: S) async throws where Element == S.Element, S: AsyncSequence {
+            for try await element in sequence {
+                try await self.write(contentsOf: CollectionOfOne(element))
```

Comment:
This seems inefficient; can't this forward on the `self.write(contentsOf:)`?

Reply:
We can't since `S` is an `AsyncSequence` here.